ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -747,16 +749,17 @@ private void checkForKeyboardEvents() {
       }
     }
 
-    private void checkForDeviceOrientationChanges() {
+    private boolean checkForDeviceOrientationChanges() {
       final int rotation =
           ((WindowManager) getContext().getSystemService(Context.WINDOW_SERVICE))
               .getDefaultDisplay()
               .getRotation();
       if (mDeviceRotation == rotation) {
-        return;
+        return false;
```

Comment:
Is there any reason not to just call `DisplayMetricsHolder.initDisplayMetrics()` here instead of returning the bool? Does it need to be called after `emitOrientationChanged`? It seems like all these methods have lots of side effects anyway

Reply:
From my testing, it seems that it's less important if ```DisplayMetricsHolder.initDisplayMetrics()``` is called before/after ```emitOrientationChanged()```. What is the most critical is that it gets called before ```checkForDeviceDimensionsChanges()```, because that's where the business logic is currently running with stale state in some scenarios (seems to be related to navigating away from the activity and then coming back).

That being said, it's totally plausible that we _do also_ need to refresh the state before calling emitOrientationChanged for other scenarios. It's possible that there are event listeners for "namedOrientationDidChange" that assume that the dimensions state holder is up to date.

It seems like the safest low risk thing to do would be to take Emily's suggestion here. @ajpaulingalls does that sound reasonable to you?

Reply:
Since `checkForDeviceOrientationChanges()` is private and currently only called by `onGlobalLayout()` you could move the call to `DisplayMetricsHolder.initDisplayMetrics()` into it.  However, I thought this was cleaner as it helped `onGlobalLayout()` describe better what it does, and its important that `checkForDeviceOrientationChanges()` happen before the `checkForDeviceDimensionsChanges()` so hvagin the logic together seemed to make sense.  Plus, as you say, its less side effecty...:)

Reply:
Hm, yeah I see. I think it makes sense to always update the display metrics if the orientation has changed, but your way makes it a little more obvious that we need to do that before updating dimensions in JS. I think the clearest way to do this would probably be to just combine checkForDeviceOrientationChanges and checkForDeviceDimensionsChanges into a single function that can enforce the order - what do you think of that? It seems like both functions are only called from this one place anyway.

I've updated the diff internally with the change I suggested above (for some reason I can't seem to sync it back to this PR so you can see it) and I've kicked off a job to merge it, since I know this is a time-sensitive fix for 0.64. If you have any more changes you can update here and I'll re-import (or we can follow up in a separate PR if it's already merged by then).

Reply:
No more changes on my side.  You are more than welcome to merge the two functions.  I agree that they work together to accomplish the shared goal, since dimensions are inexorably tied to orientation...

Reply:
Thanks @ejanzer for the expediency here. Once this commit lands in the master branch, I plan to ask for it to be cherry-picked into 0.63-stable in the [0.63.4 discussion](https://github.com/react-native-community/releases/issues/212) to support react-native@0.63 consumers who are currently impacted by this bug, and to support react-native@0.62 consumers who are blocked from upgrading to 63 because of this bug or have already rolled back to 62 to avoid it.

It looks like this PR has not been merged yet. Did the internal merge job fail? I would like to know when this commit has landed so that I can bring it up in the 0.63.4 discussion before it's too late.

Reply:
Yeah it failed. Looking into it now.

Reply:
It landed!

Reply:
Thank you Paul and Emily. This is great work!